ethstorage/p2p/discovery.go

```diff
@@ -258,16 +280,38 @@ func (n *NodeP2P) DiscoveryProcess(ctx context.Context, log log.Logger, l1ChainI
 				return
 			}
 			addrs := n.Host().Peerstore().Addrs(id)
-			log.Debug("Attempting connection", "peer", id)
+			log.Debug("Attempting connection", "peer", id, "Addr", addrs)
 			ctx, cancel := context.WithTimeout(ctx, time.Second*10)
 			err := n.Host().Connect(ctx, peer.AddrInfo{ID: id, Addrs: addrs})
 			cancel()
 			if err != nil {
-				log.Debug("Failed connection attempt", "peer", id, "err", err)
+				log.Debug("Failed connection attempt", "peer", id, "Addr", addrs, "err", err)
 			}
 		}
 	}
 
+	go func() {
+		if n.isIPSet {
+			return
+		}
+		updateLocalNodeTicker := time.NewTicker(initLocalNodeAddrInterval)
+		initialized := false
```

Comment:
Is this flag necessary?

Reply:
yes, it used to change the ticker interval. if the IP has not been set yet, it will check every 30 seconds, if the IP has been updated, it will check the IP every 10 minutes